src/test/java/com/softserveinc/dokazovi/aop/PostLoggerTest.java

```diff
@@ -0,0 +1,202 @@
+import com.softserveinc.dokazovi.repositories.PostRepository;
+import com.softserveinc.dokazovi.repositories.UserRepository;
+import com.softserveinc.dokazovi.security.UserPrincipal;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class PostLoggerTest {
+
+    @Mock
+    private LogRepository logRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private PostRepository postRepository;
+    @InjectMocks
+    private PostLogger postLogger;
+
+    private PostSaveFromUserDTO postSaveFromUserDTO;
+    private UserPrincipal userPrincipal;
+    private UserEntity userEntity;
+    private PostEntity postEntity;
+
+    @BeforeEach
+    void setUp() {
+        postSaveFromUserDTO = PostSaveFromUserDTO.builder()
+                .id(1)
+                .postStatus(0)
+                .title("testTitle").build();
+
+        userPrincipal = UserPrincipal.builder()
+                .email("test@mail.com").build();
+
+        userEntity = UserEntity.builder()
+                .firstName("testFirstName")
+                .lastName("testLastName").build();
+
+        postEntity = PostEntity.builder()
+                .status(PostStatus.DRAFT).build();
+    }
+
+    @Test
+    void saveNewPost() {
+        JoinPoint mock = Mockito.mock(JoinPoint.class);
+
+        Object[] args = new Object[]{postSaveFromUserDTO, userPrincipal};
+        when(mock.getArgs()).thenReturn(args);
+        when(userRepository.findByEmail(anyString())).thenReturn(Optional.ofNullable(userEntity));
+        postLogger.saveNewPost(mock);
+
+        verify(logRepository).save(any(LogEntity.class));
+    }
+
+    @Test
+    void updatePostSameStatus() throws Throwable {
+        ProceedingJoinPoint mock = Mockito.mock(ProceedingJoinPoint.class);
+
+        Object[] args = new Object[]{userPrincipal, postSaveFromUserDTO};
+        when(mock.getArgs()).thenReturn(args);
+        when(postRepository.getOne(anyInt())).thenReturn(postEntity);
+        when(userRepository.findByEmail(anyString())).thenReturn(Optional.ofNullable(userEntity));
+
+        postLogger.updatePost(mock);
+
+        verify(logRepository).save(any(LogEntity.class));
```

Comment:
I would add change message check for LogEntity because logRepository.save() happens any way, but the purpose of the test is to verify that the log change message was set correctly. 

This is relevant for all test-cases.